### PR TITLE
getBoundingBox method for SpawnBlockRegion and a simple testcase

### PR DESCRIPTION
**Pull request by CptCrispyCrunchy**
Needed for DynamicCities and figured it may be useful for more applications.

### CODE REVIEW

**Comment by flo** [comment]
Merged in. Although I am not sure if it is really sure if components should contain utility methods, but well when it turns out to be an issue with a new engine version it can still be removed later on.

**Comment by Cervator** [comment]
Yep Components should never contain code. The next version of the entity system will likely make that entirely impossible even. @Josharias went down a similar path some time ago and eventually started refactoring it out again but it took way more effort to fix it later.

In this case the switch should still be easy as `getBoundingBox` could simply be moved to a utility class somewhere - static method on `Region3i` ? That also makes it available to other uses :-)

File: src/main/java/org/terasology/structureTemplates/components/SpawnBlockRegionsComponent.java [hunk 1]
```diff
@@ -27,14 +28,41 @@
 /**
  * Fills regions with blocks when a entity with component receives a {@link SpawnStructureEvent}.
  */
-public class SpawnBlockRegionsComponent implements Component{
+public class SpawnBlockRegionsComponent implements Component {
     public List<RegionToFill> regionsToFill = new ArrayList<>();
 
+    public Region3i getBoundingBox() {
+        if (regionsToFill == null) {
+            return null;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/structureTemplates/components/SpawnBlockRegionsComponent.java]: Should the bounding box be of zero size in this case? Not sure if I would expect `null` to be returned for an empty template. Otherwise, add Javadoc to clarify this.

**Comment by skaldarnar** [comment]
With regard to my code comment: Add a test case for empty templates.
